### PR TITLE
[plugin.onedrive] 1.2.0

### PR DESCRIPTION
**Pull request by cguZZman**
v1.2.0 released:

Download subtitle fixed
Now supporting business accounts
xbmcvfs is used for disk operations

Previous PR issues fixed.

### CODE REVIEW

File: plugin.onedrive/addon.py [hunk 1]
```diff
@@ -278,17 +271,17 @@ def close_dialog_timeout(dialog, timeout):
         current_time = time.time()
     dialog.close()
 
-def export_folder(name, item_id, driveid, destination_folder, base_folder, directLink=None):
+def export_folder(basename, item_driveid, item_id, driveid, destination_folder, base_folder, directLink=None):
     onedrive = get_onedrive(driveid)
-    parent_folder = os.path.join(destination_folder, name)
-    if not os.path.exists(parent_folder):
+    parent_folder = destination_folder + basename + '/'
```
Comment (enen92) [inline plugin.onedrive/addon.py]: can you please use os.path.join here? It was my only comment before. Also you don't need to close and open another pull request. Just add more commits to your branch `guZZman:helix`

File: plugin.onedrive/addon.py [hunk 2]
```diff
@@ -307,29 +300,22 @@ def export_folder(name, item_id, driveid, destination_folder, base_folder, direc
...
         if is_folder:
-            export_folder(name, f['id'], driveid, parent_folder, base_folder)
+            export_folder(name, item_driveid, f['id'], driveid, parent_folder, base_folder)
         elif (('video' in f or extension in ext_videos) and content_type == 'video') or ('audio' in f and content_type == 'audio'):
-            params = {'action':'play', 'content_type': content_type, 'item_id': f['id'], 'driveid': driveid}
+            params = {'action':'play', 'content_type': content_type, 'item_driveid': item_driveid, 'item_id': f['id'], 'driveid': driveid}
             url = base_url + '?' + urllib.urlencode(params)
-            with open(os.path.join(parent_folder, name + '.strm'), 'wb') as fo:
-                fo.write(url)
+            f = xbmcvfs.File(parent_folder + name + '.strm', 'w')
```
Comment (enen92) [inline plugin.onedrive/addon.py]: please use the `with` python statement as before

File: plugin.onedrive/addon.py [hunk 3]
```diff
@@ -433,33 +435,38 @@ def report_error(e):
...
-        xbmcplugin.addDirectoryItem(addon_handle, url, list_item, True)
-        list_item = xbmcgui.ListItem(addon.getLocalizedString(32057))
-        params['action'] = 'open_simple_folder'
-        params['folder'] = 'shared'
+        params['folder'] = 'recent'
         url = base_url + '?' + urllib.urlencode(params)
         xbmcplugin.addDirectoryItem(addon_handle, url, list_item, True)
+        if content_type == 'image':
+            list_item = xbmcgui.ListItem(addon.getLocalizedString(32055))
+            params['action'] = 'open_drive_folder'
```
Comment (enen92) [inline plugin.onedrive/addon.py]: please set the media_type for your listitem

File: plugin.onedrive/addon.py [hunk 4]
```diff
@@ -433,33 +435,38 @@ def report_error(e):
...
+            params['folder'] = 'special/photos'
+            url = base_url + '?' + urllib.urlencode(params)
+            xbmcplugin.addDirectoryItem(addon_handle, url, list_item, True)
+        if content_type == 'audio':
+            list_item = xbmcgui.ListItem(addon.getLocalizedString(32056))
+            params['action'] = 'open_drive_folder'
```
Comment (enen92) [inline plugin.onedrive/addon.py]: same here

File: plugin.onedrive/resources/lib/api/onedrive.py [hunk 5]
```diff
@@ -100,49 +103,43 @@ def _make_path(self, path):
...
-            params = {}
-        if not raw_url:
-            params['access_token'] = access_token
-        return urllib.urlencode(params)
-    
+        
     def get_url(self, method, path, params=None):
         url = self._api_url+self._make_path(path)
-        if method == 'get':
+        if method == 'get' and params is not None and params != '':
```
Comment (enen92) [inline plugin.onedrive/resources/lib/api/onedrive.py]: here you could use just `and params`

File: plugin.onedrive/resources/lib/api/onedrive.py [hunk 6]
```diff
@@ -100,49 +103,43 @@ def _make_path(self, path):
...
             url = url + '?' + params
         return url
     
     def request(self, method, path, params=None, raw_url=False, retry=True):
-        url_params = self.get_url_params(params, raw_url)
+        url_params = '' if params is None else urllib.urlencode(params)
```
Comment (enen92) [inline plugin.onedrive/resources/lib/api/onedrive.py]: same here: `url_params = "" if not params else ... `

**Comment by enen92** [comment]
Thanks for the update. Some of my comments are not a blocker to merge this pull request. I would like you to, at least, implement the media types for listitem. The others are a plus.

File: plugin.onedrive/addon.py (hunk 1 above)
Comment (cguZZman) [inline plugin.onedrive/addon.py]: I made this change. 
Thanks for let me know about the pull request. Was not sure how this supposed to work.

File: plugin.onedrive/addon.py (hunk 2 above)
Comment (cguZZman) [inline plugin.onedrive/addon.py]: `with` does not support creation of files in samba, nfs or whatever the user selected as destination. So I decided to use the `xbmcvfs` library which works great so far.

File: plugin.onedrive/addon.py (hunk 3 above)
Comment (cguZZman) [inline plugin.onedrive/addon.py]: can you elaborate on this? I already set `content_type` for the listitem which is given by kodi and I just resend it across all calls (line 443). `content_type` could be image, audio or video.
I saw `media_type` definition and looks like it must be set to movie, tvshow, season, episode, musicvideo, etc., but there is no way for the addon to know this specific type apart from the `content_type`.

File: plugin.onedrive/addon.py (hunk 4 above)
Comment (cguZZman) [inline plugin.onedrive/addon.py]: same comment

File: plugin.onedrive/resources/lib/api/onedrive.py (hunk 5 above)
Comment (cguZZman) [inline plugin.onedrive/resources/lib/api/onedrive.py]: changed

File: plugin.onedrive/resources/lib/api/onedrive.py (hunk 6 above)
Comment (cguZZman) [inline plugin.onedrive/resources/lib/api/onedrive.py]: changed

**Comment by cguZZman** [comment]
Updated. Please check my comments of the reviews.
Thanks

**Comment by cguZZman** [comment]
Hey @Razzeee, I'm wondering how can I get rid of this "waiting for developer to respond" label in this PR. No more waiting for me since 3 days ago :)
Hi @enen92, I add a small fix. Hope you guys have some time to complete the review and merge the new version.
Thank you!

File: plugin.onedrive/addon.py (hunk 3 above)
Comment (enen92) [inline plugin.onedrive/addon.py]: You're right

**Pull request merged**